tests/integration/conftest.py

```diff
@@ -1078,3 +993,89 @@ def keep_centos_pointed_to_vault(shell):
         sed_repos_to_vault = r'sed -i -e "s|^\(mirrorlist=.*\)|#\1|" -e "s|^#baseurl=http://mirror\(.*\)|baseurl=http://vault\1|" /etc/yum.repos.d/CentOS-*'
         print("TESTS >>> Resetting the repos to vault")
         shell(sed_repos_to_vault, silent=True)
+
+
+class Satellite:
+    def __init__(self, key=SYSTEM_RELEASE_ENV):
+        self.shell = live_shell()
+        # Key on which upon the command is selected
+        self.key = key
+        # File containing registration commands
+        self._sat_reg_commands = dotenv_values("/var/tmp/.env_sat_reg")
+        self._sat_script_location = "/var/tmp/register_to_satellite.sh"
+        self.subman = SubscriptionManager()
+
+    def get_satellite_curl_command(self):
+        """
+        Get the Satellite registration command for the respective system.
+        """
+        if not self._sat_reg_commands:
+            pytest.fail(
+                f"The {self._sat_reg_file} either not found or empty.\
+                It is required for the satellite conversion to work."
+            )
+
+        return self._sat_reg_commands.get(self.key)
+
+    def _curl_the_satellite_script(self, curl_command):
+        assert (
+            self.shell(f"{curl_command} -o {self._sat_script_location}", silent=True).returncode == 0
+        ), "Failed to curl the satellite script to the machine."
+
```

Comment:
```suggestion
    def _curl_the_satellite_script(self, curl_command):
        assert (
            self.shell(f"{curl_command} -o {self._sat_script_location}", silent=True).returncode == 0
        ), "Failed to curl the satellite script to the machine."
        
        # This is just a mitigation of rhn-client-tools pkg obsoleting subscription-manager during upgrade
        # TODO remove when https://github.com/theforeman/foreman/pull/10280 gets merged and or foreman 3.12 is out
        # Should be around November 2024
        if "oracle" in SystemInformationRelease.distribution:
            self.shell(fr"sed -i 's/$PKG_MANAGER_UPGRADE subscription-manager/& --setopt=exclude=rhn-client-tools/' {self._sat_script_location}"
        )
```
We need to [add this](https://github.com/oamg/convert2rhel/pull/1349/files#diff-c87c5173e88bb0f60b00107f4586304f4869a976a4f998425efe3481c0db9f56L958-L964) here to beat the stubborn rhn-client-tools obsoleting subscription-manager.